File: dehaze-java/src/main/java/com/pei/dehaze/service/OrderService.java
package com.pei.dehaze.service;

import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
import com.baomidou.mybatisplus.extension.service.IService;
import com.pei.dehaze.model.entity.SysOrder;
import com.pei.dehaze.model.form.AutoRenewConfigForm;
import com.pei.dehaze.model.form.OrderCreateForm;
import com.pei.dehaze.model.form.PayRequest;
import com.pei.dehaze.model.form.RefundApplyForm;
import com.pei.dehaze.model.form.RefundAuditForm;
import com.pei.dehaze.model.query.MyOrderQuery;
import com.pei.dehaze.model.query.OrderPageQuery;
import com.pei.dehaze.model.query.RefundPageQuery;
import com.pei.dehaze.model.vo.AutoRenewConfigVO;
import com.pei.dehaze.model.vo.MyOrderVO;
import com.pei.dehaze.model.vo.OrderDetailVO;
import com.pei.dehaze.model.vo.OrderPageVO;
import com.pei.dehaze.model.vo.OrderStatsVO;
import com.pei.dehaze.model.vo.PayResult;
import com.pei.dehaze.model.vo.RefundRecordVO;

import java.time.LocalDateTime;

public interface OrderService extends IService<SysOrder> {

    PayResult create(OrderCreateForm form);

    PayResult pay(String orderNo, PayRequest request);

    void cancel(String orderNo, String reason);

    Page<MyOrderVO> listMy(MyOrderQuery query);

    OrderDetailVO getDetail(String orderNo);

    Page<OrderPageVO> getPage(OrderPageQuery query);

    OrderStatsVO getStats(LocalDateTime startTime, LocalDateTime endTime);

    void applyRefund(String orderNo, RefundApplyForm form);

    Page<RefundRecordVO> listRefunds(RefundPageQuery query);

    void approveRefund(Long refundId, RefundAuditForm form);

    void rejectRefund(Long refundId, RefundAuditForm form);

    void updateAutoRenewConfig(AutoRenewConfigForm form);

    AutoRenewConfigVO getAutoRenewConfig(Long packageId);

    void expireOrders();

    void executeRenewal();
}

File: dehaze_flutter/lib/services/order_service.dart
import 'package:dio/dio.dart';

import '../core/constants/api_constants.dart';
import '../core/network/page_result.dart';
import '../models/order_model.dart';

/// 订单服务（用户端 + 管理端）
///
/// 对齐 JS SDK OrderAPI 的全部方法。
class OrderService {
  const OrderService(this._dio);

  final Dio _dio;

  // ==================== 用户端 ====================

  /// 创建订单
  Future<MyOrderVO> create(OrderCreateForm form) async {
    final response = await _dio.post<Map<String, dynamic>>(
      ApiConstants.orders,
      data: form.toJson(),
    );
    return MyOrderVO.fromJson(
      response.data!['data'] as Map<String, dynamic>,
    );
  }

  /// 我的订单列表
  Future<PageResult<MyOrderVO>> getMyOrders(MyOrderQuery query) async {
    final response = await _dio.get<Map<String, dynamic>>(
      ApiConstants.ordersMy,
      queryParameters: query.toJson(),
    );
    final data = response.data!['data'] as Map<String, dynamic>;
    return PageResult<MyOrderVO>(
      list: (data['list'] as List<dynamic>? ?? [])
          .map((e) => MyOrderVO.fromJson(e as Map<String, dynamic>))
          .toList(),
      total: data['total'] as int? ?? 0,
    );
  }

  /// 我的订单详情
  Future<OrderDetailVO> getMyOrderDetail(int id) async {
    final response = await _dio.get<Map<String, dynamic>>(
      '${ApiConstants.ordersMy}/$id',
    );
    return OrderDetailVO.fromJson(
      response.data!['data'] as Map<String, dynamic>,
    );
  }

  /// 取消订单
  Future<void> cancelOrder(int id) async {
    await _dio.put<Map<String, dynamic>>(
      '${ApiConstants.orders}/$id/cancel',
    );
  }

  /// 发起支付
  Future<PayResult> pay(PayRequest request) async {
    final response = await _dio.post<Map<String, dynamic>>(
      '${ApiConstants.orders}/pay',
      data: request.toJson(),
    );
    return PayResult.fromJson(
      response.data!['data'] as Map<String, dynamic>,
    );
  }

  /// 查询自动续费配置
  Future<AutoRenewConfigVO> getAutoRenewConfig(int orderId) async {
    final response = await _dio.get<Map<String, dynamic>>(
      '${ApiConstants.ordersAutoRenewConfig}/$orderId',
    );
    return AutoRenewConfigVO.fromJson(
      response.data!['data'] as Map<String, dynamic>,
    );
  }

  /// 修改自动续费配置
  Future<void> updateAutoRenewConfig(AutoRenewConfigForm form) async {
    await _dio.put<Map<String, dynamic>>(
      ApiConstants.ordersAutoRenewConfig,
      data: form.toJson(),
    );
  }

  // ==================== 管理端 ====================

  /// 订单分页列表
  Future<PageResult<OrderPageVO>> getPage(OrderQuery query) async {
    final response = await _dio.get<Map<String, dynamic>>(
      ApiConstants.ordersPage,
      queryParameters: query.toJson(),
    );
    final data = response.data!['data'] as Map<String, dynamic>;
    return PageResult<OrderPageVO>(
      list: (data['list'] as List<dynamic>? ?? [])
          .map((e) => OrderPageVO.fromJson(e as Map<String, dynamic>))
          .toList(),
      total: data['total'] as int? ?? 0,
    );
  }

  /// 订单详情
  Future<OrderDetailVO> getDetail(int id) async {
    final response = await _dio.get<Map<String, dynamic>>(
      '${ApiConstants.orders}/$id',
    );
    return OrderDetailVO.fromJson(
      response.data!['data'] as Map<String, dynamic>,
    );
  }

  /// 订单统计
  Future<OrderStatsVO> getStats() async {
    final response = await _dio.get<Map<String, dynamic>>(
      ApiConstants.ordersStats,
    );
    return OrderStatsVO.fromJson(
      response.data!['data'] as Map<String, dynamic>,
    );
  }

  /// 退款分页列表
  Future<PageResult<RefundRecordVO>> getRefundPage(RefundQuery query) async {
    final response = await _dio.get<Map<String, dynamic>>(
      ApiConstants.ordersRefundsPage,
      queryParameters: query.toJson(),
    );
    final data = response.data!['data'] as Map<String, dynamic>;
    return PageResult<RefundRecordVO>(
      list: (data['list'] as List<dynamic>? ?? [])
          .map((e) => RefundRecordVO.fromJson(e as Map<String, dynamic>))
          .toList(),
      total: data['total'] as int? ?? 0,
    );
  }

  /// 退款审核
  Future<void> auditRefund(RefundAuditForm form) async {
    await _dio.put<Map<String, dynamic>>(
      ApiConstants.ordersRefunds,
      data: form.toJson(),
    );
  }
}
